### ISSUE TITLE
notification settings option text "and your own notifications" unclear

### ISSUE THREAD

**Issue by thigg** [go-gitea/gitea#25390]
### Description

It is unclear to me what the "and your own notifications" option means in the notification settings.

I think the text is improved. (And maybe also the available options could be more fine customizable as well..)

![grafik](https://github.com/go-gitea/gitea/assets/7593137/9fbc589a-d232-4075-b0ff-4cc5240fbc3f)

### Gitea Version

1.19

### Can you reproduce the bug on the Gitea demo site?

Yes

### How are you running Gitea?

self hosted debian

### Database

SQLite

**Comment by puni9869** [comment]
Please propose the changes that suites the best, I will take it forward from then, 
Ps: The current text seems fine to me.

**Comment by thigg** [comment]
If it is correct that this enables email notifications for notifications that i caused. E.g. "you pushed this commit" "you made this comment" then I'd propose
> and notifications for my own actions

I think the "and" chain is understandable. What i didnt understand what notifications are not mine.

**Comment by puni9869** [comment]
@lunny need your attention for a moment.

**Comment by lunny** [comment]
Maybe we should change these as a radiobox list and every item should have a description.

**Comment by puni9869** [comment]
We can put a lil _i_ button or tool tip in radio button title.
I propose a UI shoutly for it. I have some ideas.

**Comment by patrick-gohier-cliffs** [comment]
Bump - I'm here because one of my engineers wants to tweak email notifications and I have no idea what this option does (doesn't seem to be documented anywhere either).

**Comment by puni9869** [comment]
Apologies for the delay, I will take a look into it this weekend.

**Comment by chris-08154711** [comment]
Proposal: Discard the "enabled" keyword and label the options by what they actually contain. Swap the first two to get an order by number of mails.

| Old | New |
|---|---|
| And your own notifications | All notifications |
| Enable Email Notifications | All except own notifications |
| Only email on mention | Only on mention |
| Disable email notifications | Disable email notifications |

I am not sure about "except own" - sounds like I wont get any that are directed to myself. Maybe something like "except self inflicted" or "except triggered myself". 
Maybe a native speaker can come up with a proper wording ;)

**Comment by patrick-gohier-cliffs** [comment]
> Proposal: Discard the "Enable" keyword and label the options just by what they actually contain. Swap the first two to get an order by number of mails.
> 
> Old	New
> And your own notifications	All notifications
> Enable Email Notifications	All except own notifications
> Only email on mention	Only on mention
> Disable email notifications	Disable email notifications
> I am not sure about "except own" - sounds like I wont get any that are directed to myself. Maybe something like "except self inflicted" or "except triggered myself". Maybe a native speaker can come up with a proper wording ;)

Can you clarify what "all except own notifications" does?

**Comment by thigg** [comment]
Good proposal. I like "except triggered by me"

**Comment by buckybytes** [comment]
I think everybody found a consensus here.  No need to delay the inevitable.

**Comment by buckybytes** [comment]
For anyone who wants to take a stab at this.  The actual text values are in options/locale/locale_en-US.ini

`
email_notifications.enable = Enable Email Notifications
email_notifications.onmention = Only Email on Mention
email_notifications.disable = Disable Email Notifications
email_notifications.submit = Set Email Preference
email_notifications.andyourown = And Your Own Notifications
`

I made an attempt, but the maintainers wanted the variable names changed as well.  I attempted to find all the occurrences of the two that needed changing, but was unsuccessful.  The "AndYourOwn" and "Enabled" language is used in other places as well.

**Comment by chris-08154711** [comment]
Btw I tested "Any Your Own" a little these days (due to "Only on Mention" not working as expected, see below) and it seems it writes way to many mails. Feels more or less the same as "Enabled". I am getting a lots of mails on PRs and stuff that I am really not involved with. Please check your definition/the expected behavior of "And Your Own" and fix if needed ;)

Also "Only Email on Mention" does NOT include the notification when a reviewer has approved MY PullRequest. But that would be really helpful at this point.

Regards

**Comment by JVimes** [comment]
I did not understand the and-chain. The following would have worked for me.

_Email on:_
- All actions
- All but my actions
- @ mention only
- Nothing

(I'm not sure the use case for getting emailed on things I did.)

**Comment by chris-08154711** [comment]
We found out that the spam everyone received with the "enabled" setting was caused by the gitea setting AUTO_WATCH_NEW_REPOS, which is by default set to true. So every developer was watching every repository and therefore receiving a lot of unwanted notifications.
Maybe consider shipping gitea with this setting false by default ;)

**Comment by brasky** [comment]
> Proposal: Discard the "Enable" keyword and label the options just by what they actually contain. Swap the first two to get an order by number of mails.
> 
> Old	New
> And your own notifications	All notifications
> Enable Email Notifications	All except own notifications
> Only email on mention	Only on mention
> Disable email notifications	Disable email notifications
> I am not sure about "except own" - sounds like I wont get any that are directed to myself. Maybe something like "except self inflicted" or "except triggered myself". Maybe a native speaker can come up with a proper wording ;)

As a native english speaker this new proposal makes a lot more sense to me than the existing. I am here because I was confused looking at the options.

**Comment by mletterle** [comment]
Yes, this deserves more attention. I agree with @brasky and a number of my teammates do as well. I mean, we got what we paid for, so maybe I'll roll up my sleeves but it is a bit sad this has been open so long...
